### PR TITLE
converts the last c style for loops (that arent required to be c style) into the better for loop

### PR DESCRIPTION
**Pull request by SuperNovaa41**
There were a few of these lying around so i figured I would convert them.
Theres a couple I didn't touch because they had comments saying that that style was required, if there are any in here that aren't commented and I shouldn't touch let me know!

### CODE REVIEW

File: code/game/objects/items/plushes.dm [hunk 1]
```diff
@@ -79,14 +79,12 @@
...
-	var/i
 	var/obj/item/toy/plush/P
-	for(i=1, i<=scorned.len, i++)
+	for (var/i in 1 to scorned.len)
 		P = scorned[i]
 		P.bad_news(src)
 	scorned = null
-
-	for(i=1, i<=scorned_by.len, i++)
+	for (var/i in 1 to scroned_by.len)
```
Comment (san7890) [inline code/game/objects/items/plushes.dm:87]:
```suggestion
	for (var/i in 1 to scorned_by.len)
```

This is why yo code ain't compiling player

Comment (SuperNovaa41) [reply code/game/objects/items/plushes.dm:87]: Lol

File: code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm [hunk 2]
```diff
@@ -139,7 +139,7 @@
 	if(candidates.len < required_applicants)
 		mode.executed_rules -= src
 		return
-	for (var/i = 1, i <= required_candidates, i++)
+	for (var/i in 1 to required_candidates)
```
Comment (optimumtact) [inline code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm:142]: required_candidates.len? or length

Comment (optimumtact) [reply code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm:142]: oh it's a nubmer never mind

**Comment by Mothblocks** [comment]
most of these could just iterate over the list itself. they only use index to get the element

**Comment by SuperNovaa41** [comment]
ill do this later

**Pull request closed without merging**